bika/lims/utils/analysisrequest.py

```diff
@@ -549,3 +469,99 @@ def resolve_rejection_reasons(values):
         return [{"selected": selected, "other": other}]
 
     return []
+
+
+def do_rejection(sample, notify=None):
+    """Rejects the sample and if succeeds, generates the rejection pdf and
+    sends a notification email. If notify is None, the notification email will
+    only be sent if the setting in Setup is enabled
+    """
+    sample_id = api.get_id(sample)
+    if not sample.getRejectionReasons():
+        logger.warn("Cannot reject {} w/o rejection reasons".format(sample_id))
+        return
+
+    success, msg = doActionFor(sample, "reject")
+    if not success:
+        logger.warn("Cannot reject the sample {}".format(sample_id))
+        return
+
+    # Generate a pdf with the rejection reasons
+    pdf = get_rejection_pdf(sample)
+
+    # Attach the PDF to the sample
+    filename = "{}-rejected.pdf".format(sample_id)
+    sample.createAttachment(pdf, filename=filename)
+
+    # Do we need to send a notification email?
+    if notify is None:
+        setup = api.get_setup()
+        notify = setup.getNotifyOnSampleRejection()
+
+    if notify:
+        # Compose the email
+        mime_msg = get_rejection_mail(sample)
+        # Attach the pdf
+        mime_msg.attach(to_email_attachment(pdf))
+        # Send the email
+        send_email(mime_msg)
```

Comment:
Would you mind to refactor this and use our mail API?
https://github.com/senaite/senaite.core/blob/master/bika/lims/api/mail.py#L206

Reply:
Sorry, this one uses the API already, so maybe just the place below...

Reply:
Well, this `send_email` calls to mail API, actually